R/downloadableReactTable.R

```diff
@@ -144,37 +142,72 @@ downloadableReactTableUI <- function(id,
 #'  library(shiny)
 #'  library(periscope2)
 #'  shinyApp(ui = fluidPage(fluidRow(column(width = 12,
-#'    downloadableReactTableUI("object_id1",
-#'                             downloadtypes = c("csv", "tsv"),
-#'                             hovertext     = "Download the data here!",
-#'                             contentHeight = "300px",
-#'                             singleSelect  = TRUE)))),
-#'    server = function(input, output) {
-#'        downloadableReactTable(id         = "object_id1",
-#'                               table_data = reactiveVal(mtcars))})
+#'     downloadableReactTableUI("object_id1",
+#'                              downloadtypes = c("csv", "tsv"),
+#'                              hovertext     = "Download the data here!",
+#'                              contentHeight = "300px")))),
+#'          server = function(input, output) {
+#'            downloadableReactTable(id                = "object_id1",
+#'                                   table_data        = reactiveVal(mtcars),
+#'                                   selection_mode    = "multiple",
+#'                                   pre_selected_rows = function() {c(1, 3, 5)})})
 #'}
 #'
 #' @export
 downloadableReactTable <- function(id,
-                                   table_data) {
+                                   table_data,
+                                   selection_mode    = NULL,
+                                   pre_selected_rows = NULL) {
         shiny::moduleServer(id,
              function(input, output, session) {
                  if (is.null(table_data) || !is.function(table_data)) {
+                     message("'table_data' parameter must be a function or reactive expression.")
                      output$reactTableOutputID <- reactable::renderReactable({ NULL })
                  } else {
-                     table_react_params <- shiny::reactiveValues(table_data = NULL)
+                     table_react_params <- shiny::reactiveValues(table_data        = NULL,
+                                                                 pre_selected_rows = NULL)
+                     if (!is.null(pre_selected_rows) && !is.function(pre_selected_rows)) {
+                         message("'pre_selected_rows' parameter must be a function or reactive expression. Setting default value NULL.")
+                         pre_selected_rows <- NULL
+                     }
+
                      shiny::observe({
                          if (!is.data.frame(table_data())) {
                              table_data <- shiny::reactiveVal(data.frame(table_data()))
                          }
                          table_react_params$table_data <- table_data()
                      })
+
+                     shiny::observe({
```

Comment:
do we need two separate observes or can the observe above in line 174 be combined with this one?